ports/nrf/supervisor/qspi_flash.c

```diff
@@ -38,7 +38,35 @@
 #include "supervisor/shared/external_flash/common_commands.h"
 #include "supervisor/shared/external_flash/qspi_flash.h"
 
+#if defined(MICROPY_QSPI_OFF_WHEN_SLEEP)
+#define QSPI_ENABLE qspi_enable
+
+static void qspi_enable(void)
+{
+    if (NRF_QSPI->ENABLE) {
+        return;
+    }
+
+    nrf_qspi_enable(NRF_QSPI);
+
+    nrf_qspi_event_clear(NRF_QSPI, NRF_QSPI_EVENT_READY);
+    nrf_qspi_task_trigger(NRF_QSPI, NRF_QSPI_TASK_ACTIVATE);
+
+    uint32_t remaining_attempts = 100;
+    do {
+        if (nrf_qspi_event_check(NRF_QSPI, NRF_QSPI_EVENT_READY)) {
+            break;
+        }
+        NRFX_DELAY_US(10);
+    } while (--remaining_attempts);
+}
+
+#else
+#define QSPI_ENABLE()
```

Comment:
Suggest:  `define QSPI_ENABLE() ((void)0)` so that it is an expression with the same type in any case.

Alternately, do not introduce a macro QSPI_ENABLE() at all, and introduce a function qspi_enable which has an empty body when CIRCPY_QSPI_OFF_WHEN_SLEEP is not enabled.